snaxc/transforms/convert_dart_to_snax_stream.py

```diff
@@ -36,29 +38,82 @@ def match_and_rewrite(self, op: dart.AccessPatternOp, rewriter: PatternRewriter)
 
             # Create iterator for all dimensions of the access_mem_map that returns (stride, bound)
             # in reverse, because we work outermost -> innermost and streamers the other way around
+
+            # do not generate stride, bound pairs for irrelevant spatial dimensions
+            # all temporal dimensions are relevant for access patterns:
+            relevant: list[bool] = [True] * (pattern.num_dims - template.num_dims)
+            # relevant spatial strides have a component in the template matrix
+            relevant += template[operand].pattern.A.any(axis=0).tolist()
+
             access_iter = iter(
                 (int(pattern.A[0, i]), op.bounds.data[i].value.data)
                 for i in reversed(range(pattern.num_dims))
+                if relevant[i]
             )
 
-            # Fetch the first stride
-            stride, bound = next(access_iter)
-
             temporal_strides: list[int] = []
             spatial_strides: list[int] = []
             upper_bounds: list[int] = []
 
+            # Fetch the first stride
+            stride, bound = next(access_iter)
+
+            # TCDM takes 8 contiguous bytes minimum
+            if stride * bound == TCDM_BANK_WIDTH:
+                stride, bound = next(access_iter)
+            else:
+                stride, bound = TCDM_BANK_WIDTH, (stride * bound) // TCDM_BANK_WIDTH
+
             # fill up all spatial strides
-            for _ in [x for x in template[0].bounds if x is not None]:
-                # FIXME: provide more general solution for new spatial streamer_config
-                # configuration, this works in all current cases and layouts but is far from generally correct.
-                spatial_strides = [8]
-                stride, bound = next(access_iter, (None, None))
+            for spat_size in streamers[operand].spatial_dims:
+                spatial_strides.append(stride)
+                if bound == spat_size:
+                    # nice, strides correspond with streamer dimension
+                    stride, bound = next(access_iter)
+                elif bound < spat_size:
+                    # caution! strides of pattern don't nicely overlap with streamers, only possible
+                    # for contiguous access! Because the streamer is larger than the stride, it will
+                    # access more elements of the next stride, this is the applied stride. This should
+                    # correspond with the next stride for correct operation.
+                    assert spat_size % bound == 0
+                    applied_stride = stride * bound
+                    applied_bound = spat_size // bound
+                    next_stride, next_bound = next(access_iter)
+                    if applied_stride != next_stride:
+                        raise RuntimeError(
+                            "Non-contiguous access is not possible for this streamer configuration"
```

Comment:
Maybe something like ... "tried to apply a stride of {applied_stride}, but this is not available within ... which requires a minimum stride of {next_stride}" ? Would that make sense?